React/Views/ScrollView/RCTScrollView.m

```diff
@@ -944,7 +944,7 @@ - (void)uiManagerWillPerformMounting:(RCTUIManager *)manager
             CGPointMake(self->_scrollView.contentOffset.x + deltaX, self->_scrollView.contentOffset.y);
         if (autoscrollThreshold != nil) {
           // If the offset WAS within the threshold of the start, animate to the start.
-          if (x - deltaX <= [autoscrollThreshold integerValue]) {
+          if (x <= [autoscrollThreshold integerValue]) {
```

Comment:
This fixes `autoscrollTopThreshold`, although I ended up not using it since it will cause undesirable scroll to top when reaching the top of the list while new data is loading.